src/components/structures/MatrixChat.tsx

```diff
@@ -1379,10 +1379,13 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
                 return;
             }
 
+            Modal.closeCurrentModal("User has been logged out.");
```

Comment:
The "reason" value here is more like an event type (it's not displayed to humans), so perhaps let's pass through the event name. Also, let's add a comment for a bit more context:

```suggestion
            // A modal might have been open when we were logged out by the server
            Modal.closeCurrentModal('Session.logged_out');
```

Reply:
I believe we want to close modals in the case of soft logout as well, so please move this above the `if (...) softLogout()` block, so it happens for that path as well.

Reply:
> The "reason" value here is more like an event type (it's not displayed to humans), so perhaps let's pass through the event name. Also, let's add a comment for a bit more context:

tbh I nearly deleted the "reason" property from modal all together, it's only set twice in the entire codebase and never read, I decided not to as it's outside of the scope of this change. I've made the change you suggested.



> I believe we want to close modals in the case of soft logout as well, so please move this above the `if (...) softLogout()` block, so it happens for that path as well.

No problem, all done.

Reply:
> tbh I nearly deleted the "reason" property from Modal all together, it's only set twice in the entire codebase and never read, I decided not to as it's outside of the scope of this change.

Agreed it's rarely used today... There are a few reads though, via `onBeforeClose` handlers.